torch/export/export_schema.py

```diff
@@ -0,0 +1,326 @@
+                    # call_method and call_module are not supported, as they shouldn't apprear in the Caononical FX Graph
+                    # !!! Consider using an enum instead of string
+
+    target: str      # fully qualified name to the target, e.g. aten.add.Tensnor
+                     # !!! Consider using a structured operator name instead of string
+
+    args: List[Argument]              # args for this node
+    kwargs: List[KeywordArgument]     # kwargs for this node
+                                      # !!! Not all types in Argument are used as kwargs, e.g. TensorArgument should not be used as kwargs
+                                      # Do we want to enforce this in the schema? i.e. only allow certain types to be used as kwargs?
+
+    outputs: List[TensorArgument]   # A list of Tensor Argument returned by this node
+                                    # !!! Notice: this assumes that a node can only return Tensor(s), and not other int/float/bool types...
+
+    metadata: List[StringStringPair]
+
+
+# Maps to fx.Graph
+@dataclass
+class Graph:
+
+    inputs: List[Node]    # Maps to fx.graph's placeholder nodes.
+                          # Placeholder nodes are stored separately, to clearly distinguish them from other nodes.
+                          # !!! Consider making inputs a List[IValue] instead of List[Node], need to think about where to store the metadata for placeholder nodes
+
+    outputs: Node   # Maps to fx.graph's output node.
+                    # Output node are stored separately, to clearly distinguish them from other nodes.
+                    # A graph can only have a single output node.
+                    # !!! Consider making outputs a List[IValue] instead of a Node, need to thinking about where to store the metadata for output node
+
+    nodes: List[Node]     # maps to remaining nodes in fx.graph
+                          # Placeholder nodes and output node are not included in this list.
+
+    # Tensor values that appear in the graph
+    # They could be graph inputs, graph outputs, or intermediate values produced by nodes
+    ivalues: List[IValue]
+
+
+# Maps to fx.GraphModule
+# This the top level construct for the model
+@dataclass
+class GraphModule:
+    name: str       # A readable name for the model, potentially maps to GraphModule's self.__class__.__name__
+                    # This is not an identified for GraphModule
+
+    graph: Graph    # Only one Graph per GraphModule
+
+    metadata : List[StringStringPair]   # maps to GraphModule's meta, which is a Dict[str, Any], but we only support string key and string value.
+
+    # Need to store stateful information
+
+    # The name of the tensor will be used to bind to the IValues of Graph Inputs
+    # !!! Consider storing them in the GraphModule, or in the Graph.
+    # !!! Do we needs to store parameters and buffers separately? (Sherlock: Ideally no.)
```

Comment:
yeah we shouldnt store them separately.

Parameters with requires_grad = False are essentially buffers

Reply:
There are functional difference between buffers and parameters:

see
https://discuss.pytorch.org/t/what-is-the-difference-between-register-buffer-and-register-parameter-of-nn-module/32723
and
https://stackoverflow.com/questions/57540745/what-is-the-difference-between-register-parameter-and-register-buffer-in-pytorch

So we still need to store them separately ...